src/main/java/bio/terra/service/snapshotbuilder/query/SqlRenderContext.java

```diff
@@ -0,0 +1,44 @@
+package bio.terra.service.snapshotbuilder.query;
+
+import bio.terra.common.CloudPlatformWrapper;
+import java.util.HashMap;
+import java.util.Map;
+
+public class SqlRenderContext {
+  private final TableNameGenerator tableNameGenerator;
+  private final CloudPlatformWrapper platform;
+  private final Map<TableVariable, String> aliases = new HashMap<>();
+
+  public SqlRenderContext(TableNameGenerator tableNameGenerator, CloudPlatformWrapper platform) {
+    this.tableNameGenerator = tableNameGenerator;
+    this.platform = platform;
```

Comment:
What if platform is null? Would we still need the InvalidRenderSqlParameter exception?

Reply:
I don't think there's a way for it to be `null`. The only call to create `SqlRenderContext` is in `SnapshotBuilderService.createContext()`, and that method uses `CloudPlatformWrapper.of(dataset.getCloudPlatform())`. So it should never be `null` in practice.